File: 프로그래머스/lv1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.java
class Solution {
    public String[] solution(int n, int[] arr1, int[] arr2) {
        int[] rows = new int[arr1.length];
        for(int i=0; i<arr1.length;i++)
            rows[i] = (int)(arr1[i] | arr2[i]);
        String[] answer = new String[arr1.length];
        for(int i=0; i<arr1.length;i++){
            answer[i] = Integer.toString(rows[i],2);
            while(answer[i].length() < n)
                answer[i] = "0"+answer[i];
            
            answer[i] = answer[i].replaceAll("0"," ");
            answer[i] = answer[i].replaceAll("1","#");
        }
        return answer;
    }
}

File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.js
function solution(n, arr1, arr2) {
    const map1 = arr1.map((num)=>{
        return getRow(num, n)
    })
    const map2 = arr2.map((num)=>{
        return getRow(num, n)
    })
    
    const final = map1.map((_,rowIdx)=>{
        const row1 = map1[rowIdx]
        const row2 = map2[rowIdx]
        
        const finalRow = []
        for(let colIdx=0; colIdx<n; colIdx++)
            finalRow.push(row1[colIdx] | row2[colIdx])
        return finalRow
    })
    
    const answer = []
    const dict = {
        1: '#',
        0: ' ',
    }
    
    for(let row = 0; row < n ; row++) {
        let str = ''
        for(let col=0; col<n; col++) 
            str += dict[final[row][col]]
        answer.push(str)
    }
    
    
    return answer
}

function getRow(num, n){
    const stringify = num.toString(2).padStart(n,'0')
    return stringify.split('').map(Number)
}